src/Main.cpp

```diff
@@ -379,90 +374,90 @@ int main(int argc, char** argv)
         }
         bool enable = enabled_s == "on";
 
-        if (!PORT_S_TO_E.count(port)) {
-            printPorts();
-            return 1;
-        }
         if (!PROTOCOL_S_TO_E.count(protocol)) {
             printProtocols();
             return 1;
         }
         driver.openURI(uri);
-        driver.setPortProtocol(PORT_S_TO_E.at(port),
-                               input ? DIRECTION_INPUT : DIRECTION_OUTPUT,
-                               PROTOCOL_S_TO_E.at(protocol),
-                               enable);
+        driver.setPortProtocol(
+            port, input ? DIRECTION_INPUT : DIRECTION_OUTPUT,
+            PROTOCOL_S_TO_E.at(protocol), enable
```

Comment:
it is not a requirement to check if "protocol" is valid? (just like you did with the port)

Reply:
It is actually checked just above, and map.at throws if the key does not exist

Nonetheless, modified by creating a protocolFromString for clarity.